File: static/js/recruiters.js
async function recruiterSignup() {

    const name = document.getElementById("name").value
    const email = document.getElementById("email").value
    const phone = document.getElementById("phone").value
    const password = document.getElementById("password").value
    const state = document.getElementById("state").value
    const company = document.getElementById("company").value

    const response = await fetch("/api/recruiter-signup", {

        method: "POST",

        headers: {
            "Content-Type": "application/json"
        },

        body: JSON.stringify({
            name: name,
            email: email,
            phone: phone,
            password: password,
            state: state,
            company: company
        })

    })

    const data = await response.json()

    if (data.status === "success") {

        alert("Recruiter Signup Successful")

        window.location.href = "/login"

    } else {

        alert(data.message)

    }

}

File: backend/recruiters.py
from fastapi import APIRouter
from pydantic import BaseModel
from database import recruiters_collection

router = APIRouter()


class RecruiterSignup(BaseModel):
    name: str
    email: str
    phone: str
    password: str
    state: str
    company: str


@router.post("/api/recruiter-signup")
def recruiter_signup(data: RecruiterSignup):

    existing = recruiters_collection.find_one({"email": data.email})

    if existing:
        return {"status": "error", "message": "Email already exists"}

    recruiters_collection.insert_one(data.dict())

    return {"status": "success", "message": "Recruiter registered successfully"}


@router.get("/api/recruiters")
def get_recruiters():

    recruiters = list(recruiters_collection.find({}, {"_id": 0}))

    return recruiters
